src/main/java/net/dv8tion/jda/core/entities/impl/JDAImpl.java

```diff
@@ -297,6 +298,36 @@ public User getUserById(String id)
     }
 
     @Override
+    public RestAction<User> retrieveUser(String id)
+    {
+        if (accountType != AccountType.BOT)
+            throw new AccountTypeException(AccountType.BOT);
+        Args.notNull(id, "User id");
+        // check cache
+        User user = this.getUserById(id);
+        if (user == null)
+            user = this.fakeUsers.get(id);
```

Comment:
Don't check the fake user cache in this situation. Unlike the live cache which is always up to date, the fake user cache may not be up to date.